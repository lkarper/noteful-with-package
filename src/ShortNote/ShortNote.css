li {
    border: 1px solid black;
    margin: 5px;
    display: flex;
    flex-direction: column;
    padding: 5px;
}

li button {
    max-width: 30%;
}

li a {
    text-decoration: none;
    color: black;
    font-size: 32px;
}

li a:hover, li a:focus {
    text-decoration: underline;
}