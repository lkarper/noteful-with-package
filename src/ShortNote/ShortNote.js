import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import NotesContext from '../NotesContext/NotesContext';
import config from '../config';
import './ShortNote.css';

class ShortNote extends Component {

    state = {
        error: null,
    }

    static contextType = NotesContext;

    static defaultProps = {
        note: {
            name: "Placeholder name",
            id: 0, 
            date_modified: (new Date()).toJSON(), 
            folder: 0, 
            content: "Lorem ipsum...",
        },
    }

    deleteNoteRequest = (noteId, cb) => {
        fetch(`${config.API_URL}/notes/${noteId}`, {
            method: 'DELETE',
            headers: {
                'content-type': 'application/json',
                'Authorization': `Bearer ${config.API_KEY}`
            },
        })
        .then(response => {
            if (!response.ok) {
                return response.json().then(error => {
                    throw error;
                })
            }
        })
        .then(data => {
            cb(noteId);
        })
        .catch(error => {
            console.log(error);
            this.setState({ error: error.message });
        });
    }

    render() {
        const error = this.state.error;
        const errorHTML = (
            <div>
                <h2>Could not delete note: {error}.</h2>
                <p>Check your connection and try again.</p>
            </div>
        );

        const { note } = this.props;
        const date = new Date(note.date_modified);

        return (
            <li>
                <Link to={`/note/${note.id}`}>
                    {note.name}
                </Link>
                <p>{`Last modified on: ${date.toString()}`}</p>
                <button 
                    type="button"
                    onClick={() => this.deleteNoteRequest(note.id, this.context.deleteNote)}    
                >Delete Note</button>
                {error ? errorHTML : ''}
            </li>
        );
    }
}

ShortNote.propTypes = {
    note: PropTypes.shape({
        name: PropTypes.string,
        id: PropTypes.number,
        date_modified: PropTypes.string,
        folder: PropTypes.number,
        content: PropTypes.string,
    }).isRequired,
}

export default ShortNote;