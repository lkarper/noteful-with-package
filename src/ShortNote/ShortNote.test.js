import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ShortNote from './ShortNote';

describe('ShortNote component', () => {
    it('renders the UI without crashing', () => {
        const wrapper = mount(
        <BrowserRouter>
            <ShortNote />
        </BrowserRouter>);
        wrapper.unmount();
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(
            <BrowserRouter>
                <ShortNote />
            </BrowserRouter>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });

    it('renders an error message when the delete request fails', () => {
        const wrapper = mount(
            <BrowserRouter>
                <ShortNote />
            </BrowserRouter>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });

});
