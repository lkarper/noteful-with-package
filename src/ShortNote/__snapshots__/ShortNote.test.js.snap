// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortNote component renders an error message when the delete request fails 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ShortNote
      note={
        Object {
          "content": "Lorem ipsum...",
          "date_modified": "2020-06-29T16:02:09.859Z",
          "folder": 0,
          "id": 0,
          "name": "Placeholder name",
        }
      }
    >
      <li>
        <Link
          to="/note/0"
        >
          <LinkAnchor
            href="/note/0"
            navigate={[Function]}
          >
            <a
              href="/note/0"
              onClick={[Function]}
            >
              Placeholder name
            </a>
          </LinkAnchor>
        </Link>
        <p>
          Last modified on: Mon Jun 29 2020 12:02:09 GMT-0400 (Eastern Daylight Time)
        </p>
        <button
          onClick={[Function]}
          type="button"
        >
          Delete Note
        </button>
      </li>
    </ShortNote>
  </Router>
</BrowserRouter>
`;

exports[`ShortNote component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ShortNote
      note={
        Object {
          "content": "Lorem ipsum...",
          "date_modified": "2020-06-29T16:02:09.859Z",
          "folder": 0,
          "id": 0,
          "name": "Placeholder name",
        }
      }
    >
      <li>
        <Link
          to="/note/0"
        >
          <LinkAnchor
            href="/note/0"
            navigate={[Function]}
          >
            <a
              href="/note/0"
              onClick={[Function]}
            >
              Placeholder name
            </a>
          </LinkAnchor>
        </Link>
        <p>
          Last modified on: Mon Jun 29 2020 12:02:09 GMT-0400 (Eastern Daylight Time)
        </p>
        <button
          onClick={[Function]}
          type="button"
        >
          Delete Note
        </button>
      </li>
    </ShortNote>
  </Router>
</BrowserRouter>
`;
